### PR TITLE
Refector Code Quality Issues

### PR DESCRIPTION
**Pull request by ankitdobhal**
## Description

Hey 👋 , I work at DeepSource, I ran the DeepSource static analyzer on the forked copy of this repo and found some [interesting code quality issues](https://deepsource.io/gh/ankitdobhal/cocotb/issues). This PR fixes a few of them.

## Following are the changes-

- Remove implicit `object` from the base class
- Use literal syntax to create data structure
- Remove unnecessary return statement
- Add .deepsource.toml file for continuous analysis on bug risks/performance/code-quality issues on new changes

_Note: I've fixed a few issues for now. You can find and fix other issues [here](https://deepsource.io/gh/ankitdobhal/cocotb/issues)._

## Type of change

- [x] Antipattern
- [ ] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: cocotb/scheduler.py [hunk 1]
```diff
@@ -453,24 +451,13 @@ def _event_loop(self, trigger):
                     if _debug:
                         self.log.debug("Scheduled coroutine %s" % (coro._coro.__qualname__))
 
-                    # remove our reference to the objects at the end of each loop,
-                    # to try and avoid them being destroyed at a weird time (as
-                    # happened in gh-957)
-                    del coro
-
```
Comment (eric-wieser) [inline cocotb/scheduler.py:460]: This `del`, and many of the others, is important.

File: tests/test_cases/test_compare/test_compare.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 from cocotb.triggers import RisingEdge, FallingEdge
 
 
-class Testbench(object):
+class Testbench():
```
Comment (eric-wieser) [inline tests/test_cases/test_compare/test_compare.py:14]: This change is good, but I'd be inclined to drop the `()` everywhere too.

I also think that #2423 probably does this for us anyway.

**Comment by imphil** [comment]
I'm sorry, but this PR feels a bit like an ad for DeepSource. We can discuss if it brings something useful for us, but not in a PR. And as @eric-wieser said, the Python2 -> Python3 issues should be solved in #2423

File: cocotb/scheduler.py (hunk 1 above)
Comment (ankitdobhal) [reply cocotb/scheduler.py]: @eric-wieser reverted the changes.

File: cocotb/wavedrom.py [hunk 3]
```diff
@@ -168,7 +168,6 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         for sig in self._signals:
             sig.clear()
         self.disable()
-        return None
```
Comment (ktbarrett) [inline cocotb/wavedrom.py:171]: In fact. I think this is the only change not covered by #2423.

**Review by ktbarrett — COMMENTED** [review]
Looking at the report generated there are other issues that could be addressed like importing `cocotb.simulator` multiple times in `_initialize_testbench`. The definition of `__iter__` on the decorators (which I noted before as being wrong). Lots of instances of unused variables rather than using `_`. Is there a reason you didn't include any of those changes?

**Comment by cmarqu** [comment]
E.g. pylint still also has lots of things to report that may be fishy in the code.
flake8 found unused variables in the tests and examples, but I feel having them named makes the code "more didactic", so our flake8 config ignores these reports for those files.

**Comment by ktbarrett** [comment]
Seeing as the current set of code changes are not really useful, I guess this PR can be more about whether we want to add DeepSource support. I can't tell from their website whether running it solely in CI is considered a single "team member" or not. If there is any restriction on our use in CI, I don't want it in the repo.

**Comment by ktbarrett** [comment]
Gonna say no. If someone else agrees, go ahead an close this PR.

**Pull request closed without merging**